executor/analyze.go

```diff
@@ -345,31 +350,32 @@ func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultsCh chan<-
 				Job: task.job,
 			}
 		}
-		e.wg.Done()
-		if isCloseChanThread {
-			e.wg.Wait()
-			close(resultsCh)
-		}
 	}()
 	for {
-		var ok bool
-		task, ok = <-taskCh
-		if !ok {
-			break
-		}
-		task.job.Start()
-		switch task.taskType {
-		case colTask:
-			resultsCh <- analyzeColumnsPushdown(task.colExec)
-		case idxTask:
-			resultsCh <- analyzeIndexPushdown(task.idxExec)
-		case fastTask:
-			resultsCh <- analyzeFastExec(task.fastExec)
-		case pkIncrementalTask:
-			resultsCh <- analyzePKIncremental(task.colIncrementalExec)
-		case idxIncrementalTask:
-			resultsCh <- analyzeIndexIncremental(task.idxIncrementalExec)
+		select {
+		case task := <-taskCh:
+			task.job.Start()
+			switch task.taskType {
+			case colTask:
+				resultsCh <- analyzeColumnsPushdown(task.colExec)
+			case idxTask:
+				resultsCh <- analyzeIndexPushdown(task.idxExec)
+			case fastTask:
+				resultsCh <- analyzeFastExec(task.fastExec)
+			case pkIncrementalTask:
+				resultsCh <- analyzePKIncremental(task.colIncrementalExec)
+			case idxIncrementalTask:
+				resultsCh <- analyzeIndexIncremental(task.idxIncrementalExec)
+			}
+		default:
+			select {
+			case <-exitCh:
```

Comment:
This exitCh is closed just after we put all tasks into the taskCh. When will the case that `we lost data when exitCh is closed` would happen?

Reply:
```
select{
   case <-task:
   case <-exitch:
}
```

if ```exitch``` is closed and it still has the task in the task channel, ```select``` maybe run the ```exitch``` branch and  ignore the ones in the task channel.

Reply:
removed